mantle/platform/api/azure/instance.go

```diff
@@ -156,7 +156,11 @@ func (a *API) CreateInstance(name, userdata, sshkey, resourceGroup, storageAccou
 
 	vmParams := a.getVMParameters(name, userdata, sshkey, fmt.Sprintf("https://%s.blob.core.windows.net/", storageAccount), ip, nic)
 
-	_, err = a.compClient.CreateOrUpdate(resourceGroup, name, vmParams, nil)
+	cancel := make(chan struct{})
+	time.AfterFunc(5*time.Minute, func() {
+		cancel <- struct{}{}
+	})
+	_, err = a.compClient.CreateOrUpdate(resourceGroup, name, vmParams, cancel)
```

Comment:
What error is returned in case of timeout?  Should we wrap the message for debuggability?

Reply:
Digging into the code, this is what seems to be returned in case of error (in `autorest/sender.go`):
```golang
// AfterDelay returns a SendDecorator that delays for the passed time.Duration before
// invoking the Sender. The delay may be terminated by closing the optional channel on the
// http.Request. If canceled, no further Senders are invoked.
func AfterDelay(d time.Duration) SendDecorator {
	return func(s Sender) Sender {
		return SenderFunc(func(r *http.Request) (*http.Response, error) {
			if !DelayForBackoff(d, 0, r.Cancel) {
				return nil, fmt.Errorf("autorest: AfterDelay canceled before full delay")
			}
			return s.Do(r)
		})
	}
}
```
I think wrapping the error with just a `CreateOrUpdate:` will be useful.